platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -124,52 +219,58 @@ CGContextHandle context(CGBitmapContextCreate(
     if (!context) {
         throw std::runtime_error("CGBitmapContextCreate failed");
     }
-
-    CFStringRef keys[] = { kCTFontAttributeName };
-    CFTypeRef values[] = { font };
-
-    CFDictionaryRefHandle attributes(
-        CFDictionaryCreate(kCFAllocatorDefault, (const void**)&keys,
-            (const void**)&values, sizeof(keys) / sizeof(keys[0]),
-            &kCFTypeDictionaryKeyCallBacks,
-            &kCFTypeDictionaryValueCallBacks));
-
-    CFAttributedStringRefHandle attrString(CFAttributedStringCreate(kCFAllocatorDefault, *string, *attributes));
-    CTLineRefHandle line(CTLineCreateWithAttributedString(*attrString));
     
-    // Start drawing a little bit below the top of the bitmap
-    CGContextSetTextPosition(*context, 0.0, 5.0);
+    CFArrayRef glyphRuns = CTLineGetGlyphRuns(*line);
+    CTRunRef glyphRun = (CTRunRef)CFArrayGetValueAtIndex(glyphRuns, 0);
+    CFRange wholeRunRange = CFRangeMake(0, CTRunGetGlyphCount(glyphRun));
+    CGSize advances[wholeRunRange.length];
+    CTRunGetAdvances(glyphRun, wholeRunRange, advances);
+    metrics.advance = std::round(advances[0].width);
+    
+    // Mimic glyph PBF metrics.
+    metrics.left = Glyph::borderSize;
+    metrics.top = -1;
```

Comment:
I replaced the iOS/macOS test fixture with the Qt test fixture, which does a much better job of matching Core Text’s actual typesetting behavior with respect to the baseline and kerning. I had to round the advances, because mbgl doesn’t support subpixel text rendering, which has been standard on Apple platforms since iOS 2.0 and macOS 10.5 Leopard.

Old expected | Actual | New expected
----|----|----
![old](https://user-images.githubusercontent.com/1231218/75632161-392e6380-5bae-11ea-8958-8b65ea30a612.png) | ![actual](https://user-images.githubusercontent.com/1231218/79942515-dde55880-841b-11ea-95c7-76d590b3c0ce.png) | ![new](https://user-images.githubusercontent.com/1231218/79942502-d920a480-841b-11ea-9650-3790dea8cea5.png)